zerver/tests/test_reactions.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+import ujson
+from typing import Any, Dict, List
+from six import string_types
+
+from zerver.lib.test_helpers import tornado_redirected_to_list, get_display_recipient
+from zerver.lib.test_classes import ZulipTestCase
+from zerver.models import get_user_profile_by_email
+
+class ReactionEmojiTest(ZulipTestCase):
+    def test_missing_emoji(self):
+        # type: () -> None
+        """
+        Sending reaction without emoji fails
+        """
+        sender = 'hamlet@zulip.com'
+        result = self.client_post('/api/v1/reactions', {'message_id': 1},
+                                  **self.api_auth(sender))
+        self.assert_json_error(result, 'Missing \'emoji\' argument')
+
+    def test_empty_emoji(self):
+        # type: () -> None
+        """
+        Sending empty emoji fails
+        """
+        sender = 'hamlet@zulip.com'
+        result = self.client_post('/api/v1/reactions', {'message_id': 1, 'emoji': ''},
+                                  **self.api_auth(sender))
+        self.assert_json_error(result, "Emoji name can't be empty")
+
+    def test_invalid_emoji(self):
+        # type: () -> None
+        """
+        Sending invalid emoji fails
+        """
+        sender = 'hamlet@zulip.com'
+        result = self.client_post('/api/v1/reactions', {'message_id': 1, 'emoji': 'foo'},
+                                  **self.api_auth(sender))
+        self.assert_json_error(result, 'Emoji does not exist')
+
+    def test_valid_emoji(self):
+        # type: () -> None
+        """
+        Reacting with valid emoji succeeds
```

Comment:
maybe edit this docstring to "Reacting with valid realm emoji succeeds", and then add a much simpler test for the case of a standard (non-realm-specific) emoji.